### DISCUSSION TITLE
Improved clarity of page images

### DISCUSSION THREAD

**Discussion by deahhh** [pymupdf/PyMuPDF#1862] [Q&A] [answered]
I'm not sure if this is a problem.
Does the library only extract the data contained in the pdf file?
Does the pdf have a protocol for blurring the data and displaying it?
Evince can display very clear content (not reproducible, probably images, using fitz can get the image of the base64 string), but use fitz to extract the page image file is very blurred, is Evince made digital filtering process?

I'm very sorry, I asked Evince, but I didn't get a reply, so I didn't have a solution, so I thought I'd try my luck here.

Does this library have plans for super-resolution processing of pdf displays?

**Comment by JorjMcKie** [comment]
This is no issue, but a question, so I converted your post.

Several comments in your post are hard to understand:
* what do you mean with "blurred", "blurring" data?
* how comes base64 encoding into play here?
* do you mean **_making_** an image from a page or **_extracting_** images **_shown_** on a page?

I am confused.

**Reply by deahhh** [reply]
> This is no issue, but a question, so I converted your post.
> 
> Several comments in your post are hard to understand:
> 
>     * what do you mean with "blurred", "blurring" data?
> 
>     * how comes base64 encoding into play here?
> 
>     * do you mean **_making_** an image from a page or **_extracting_** images **_shown_** on a page?
> 
> 
> I am confused.

Thank you for your reply.
The text(I see) in my pdf file is not copyable. I guess it is composed of images. And, with pyMuPdf parsing out of the page is indeed no text, but there is a file string, can be opened by opencv. But this picture is very unclear. This is very consistent with the size of the parsed image 300x600 (such a size how can be clearly visible).
The problem is that Evince open this pdf file text is very clear.

I just don't know the where is the differ come from, one is clear while the other is blurred, unreadable. Does it come from pdf protocols which limit it to show? Or come from Evince who may be doing extra work to make it readable?

That's all

**Reply by JorjMcKie** [reply]
how do you create this image ("file string")?

**Reply by JorjMcKie** [reply]
Your text need not necessarily be an image. It also may be so-called line art: elementary drawing operations (lines, curves, ...) used to paint characters.
Like a capital "D" could be drawn by a "|" followed by a left-open semi-circle. You get the point.

**Reply by JorjMcKie** [reply]
> Does it come from pdf protocols which limit it to show? Or come from Evince who may be doing extra work to make it readable?

No, nothing like that of course.
It is probably caused by the way **_you_** created that image.

**Reply by deahhh** [reply]
> how do you create this image ("file string")?

it's not created by me, but read from the file with pymupdf.
it's image file, in string.

**Reply by deahhh** [reply]
> Does it come from pdf protocols which limit it to show? Or come from Evince who may be doing extra work to make it readable?

nice, it must be some tech used by Evince, which make the showing in it more clear and readable.

**Reply by JorjMcKie** [reply]
what do you mean 'not created by me'? Please tell me the instruction that delivered the image

**Reply by JorjMcKie** [reply]
> it's not created by me, but **_read from the file with pymupdf_**.
> it's image file, in string.

Again my question: what did you do to read from the file?
* `doc.extract_image()`?
* `page.get_pixmap()`?
* or what else?

**Reply by deahhh** [reply]
> 

Thanks again!
`page.get_pixmap()`
I should give this information at beginning. So sorry.

The [pdf](https://drive.google.com/file/d/1Jr7zdJR3Y3UqV30V_rmoSV2J0ijcty2-/view?usp=sharing)

code:
```Python
import fitz
import re
import os

file_path = r'1.pdf'       # PDF 
dir_path = r'./images'  

def pdf2image1(pdf_path, pic_path):
    checkIM = r"/Subtype(?= */Image)"
    doc = fitz.open(pdf_path)

    for i in range(1, doc.page_count):
        pg = doc.load_page(i).get_svg_image()
        
        pix = doc[i].get_pixmap()
        pix.save(os.path.join(pic_path, f"page_{i}.png"))

pdf2image1(file_path, dir_path)
```

**Reply by deahhh** [reply]
@JorjMcKie 
Thank you very much!
I've got the origin image now. Inspired by your [example code](https://github.com/pymupdf/PyMuPDF/discussions/1825)

**Comment by JorjMcKie** [comment, marked as answer]
> I should give this information at beginning. So sorry.

Indeed you should!

If you look at the documentation, you will find **_several options_** to increase the resolution of a pixmap. The easiest to use is parameter `dpi`: `page.get_pixmap(dpi=300)` for example creates an image with 300 dots per square inch.

**Reply by JorjMcKie** [reply]
And another comment:
When using `page.get_pixmap()`, you **_are_** creating an image. It is now clear that your previous statement "it's not created by me, but read from the file with pymupdf" was simply wrong.

**Reply by deahhh** [reply]
Yes. 
It's the name `get_xxx` made me consider `get_pixmap` as nothing to do with `create`. 
Stereotypes of thinking made me consider that there is a `show` do something like `draw_pixmap`.

Thanks again!
I think it's great to use `pymupdf` do something interesting.
